src/main/java/hudson/plugins/timestamper/io/TimeShiftsReader.java

```diff
@@ -72,7 +71,7 @@
    */
   Optional<Long> getTime(long timestampEntry) throws IOException {
     if (timeShifts == null) {
-      timeShifts = ImmutableMap.copyOf(readTimeShifts());
+      timeShifts = Collections.unmodifiableMap(readTimeShifts());
```

Comment:
this is not a direct replacement.  the returned map can change (even though it is unmodifiable) as it reasds through to the underlying map.

based on the actual implementation of `readTimeShifts` it is ok here though.

Reply:
Yes, not an issue based on the actual implementation of `readTimeShifts`.